main/vstring.c

```diff
@@ -90,7 +83,7 @@ extern vString *vStringNew (void)
 extern void vStringPut (vString *const string, const int c)
 {
 	if (string->length + 1 == string->size)  /*  check for buffer overflow */
-		vStringAutoResize (string);
+		vStringResize (string, string->size * 2);
```

Comment:
I'd rather just ask for the extra byte required here, and let the `vStringResize()` implementation do the doubling.  Here it's kind of misleadingly suggesting `vStringResize()` gives the exact requested size, and as it's not the case, adds an extra location where the doubling happens, which means it'd have to be changed if resizing policy ever changes.